examples/multisig_plain/lib.rs

```diff
@@ -0,0 +1,837 @@
+            assert!(contract.is_owner.get(&accounts.alice).is_none());
+            assert!(contract.is_owner.get(&accounts.django).is_some());
+            assert_eq!(test::recorded_events().count(), 2);
+        }
+
+        #[test]
+        #[should_panic]
+        fn replace_owner_existing_fails() {
+            let accounts = default_accounts();
+            let mut contract = build_contract();
+            set_from_wallet();
+            contract.replace_owner(accounts.alice, accounts.bob);
+        }
+
+        #[test]
+        #[should_panic]
+        fn replace_owner_nonexisting_fails() {
+            let accounts = default_accounts();
+            let mut contract = build_contract();
+            set_from_wallet();
+            contract.replace_owner(accounts.django, accounts.frank);
+        }
+
+        #[test]
+        #[should_panic]
+        fn replace_owner_permission_denied() {
+            let accounts = default_accounts();
+            let mut contract = build_contract();
+            set_from_owner();
+            contract.replace_owner(accounts.alice, accounts.django);
+        }
+
+        #[test]
+        fn change_requirement_works() {
+            let mut contract = build_contract();
+            assert_eq!(*contract.requirement.get(), 2);
+            set_from_wallet();
+            contract.change_requirement(3);
+            assert_eq!(*contract.requirement.get(), 3);
+            assert_eq!(test::recorded_events().count(), 1);
+        }
+
+        #[test]
+        #[should_panic]
+        fn change_requirement_too_high() {
+            let mut contract = build_contract();
+            set_from_wallet();
+            contract.change_requirement(4);
+        }
+
+        #[test]
+        #[should_panic]
+        fn change_requirement_zero_fails() {
+            let mut contract = build_contract();
+            set_from_wallet();
+            contract.change_requirement(0);
+        }
+
+        #[test]
+        fn submit_transaction_works() {
+            submit_transaction();
+        }
+
+        #[test]
+        #[should_panic]
+        fn submit_transaction_noowner_fails() {
+            let mut contract = build_contract();
+            set_from_noowner();
+            contract.submit_transaction(Transaction::change_requirement(1));
+        }
+
+        #[test]
+        #[should_panic]
+        fn submit_transaction_wallet_fails() {
+            let mut contract = build_contract();
+            set_from_wallet();
+            contract.submit_transaction(Transaction::change_requirement(1));
+        }
+
+        #[test]
+        fn cancel_transaction_works() {
+            let mut contract = submit_transaction();
+            set_from_wallet();
+            contract.cancel_transaction(0);
+            assert_eq!(contract.transactions.len(), 0);
+            assert_eq!(test::recorded_events().count(), 3);
+        }
+
+        #[test]
+        fn cancel_transaction_nonexisting() {
```

Comment:
So it doesn't panic when I cancel a non-existing transaction?

Reply:
Yep. It just does nothing. The code is just a little simpler that way and uses less gas in the common case (at least I think it does).